### ISSUE TITLE
Conversion to fiat in transaction history uses current conversion rate

### ISSUE THREAD

**Issue by ShadyIIV** [walleth/walleth#21]
#### Issue description

All fiat numbers in the transaction history get converted according to todays conversion rate.

#### Steps to reproduce the issue

1. Make a transaction
2. Wait some days
3. See the FIAT value change

#### What's the expected result?

The app should use the conversion rate from the date of the transaction. If a user sends 10 € in ETH to a friend for a cinema ticket and ETH loses 30% of it's value then it should still say that I transfered 10€, because that's how much it was worth.
 
#### What's happening:

The numbers constantly get converted again according to todays course and therefor always change, making it confusing and difficult to see what the transaction were.

**Comment by ligi** [comment]
Thanks for the suggestion - but I am not sure if this is a good Idea. This could get really confusing for users and might get problematic as I need to find a source for historical exchange rates until genesis. As WALLETH can be used as a watch only wallet for any address one could use an address that made a transaction short after genesis.
That said - thanks for the Input - I will think about it and talk to other users about it.

**Comment by ShadyIIV** [comment]
Right, didn't think much about the implementation. Maybe that's why some apps that I saw don't show the FIAT value in the transaction history. 

> This could get really confusing for users 

Isn't that expected behaviour from how exchange rates usually work? If the app is used for day-to-day payments, but the rate fluctuates then the FIAT info becomes at best a fun information ("look, half a year ago I payed 1200€ for a concert ticket") and at worst meaningless/confusing ("Hey, you owed me 10$, why did you sent just $7 last Sunday?").

> I need to find a source for historical exchange rates until genesis

A option could be to only have the FIAT value listed in the transaction if it was done since the address was added to the app. (Or maybe have the fiat value maybe only in the transaction details and titled: "Value today" to avoid confusion.)
